internal/engine/upper.go

```diff
@@ -412,8 +413,8 @@ func handleBuildCompleted(ctx context.Context, engineState *store.EngineState, c
 		}
 
 		bestPod := ms.MostRecentPod()
-		if store.AfterOrEqual(bestPod.CreatedAt.Time, bs.StartTime) ||
-			bestPod.UpdateStartedAt.Time.Equal(bs.StartTime) {
+		if timecmp.AfterOrEqual(bestPod.CreatedAt, bs.StartTime) ||
+			timecmp.Equal(bestPod.UpdateStartedAt, bs.StartTime) {
```

Comment:
This was the case that motivated me to do this -- build times are stdlib times while the Pod times are API times, so this comparison was problematic (it started breaking tests with more Pod changes).